backend/common/src/main/java/ai/verta/modeldb/common/CommonApp.java

```diff
@@ -0,0 +1,113 @@
+package ai.verta.modeldb.common;
+
+import ai.verta.modeldb.common.config.Config;
+import ai.verta.modeldb.common.futures.FutureGrpc;
+import io.grpc.Server;
+import io.prometheus.client.exporter.MetricsServlet;
+import io.prometheus.client.hotspot.DefaultExports;
+import io.prometheus.jmx.BuildInfoCollector;
+import io.prometheus.jmx.JmxCollector;
+import java.io.BufferedReader;
+import java.io.File;
+import java.io.FileReader;
+import java.util.Optional;
+import java.util.concurrent.Executor;
+import java.util.concurrent.atomic.AtomicBoolean;
+import javax.management.MalformedObjectNameException;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.springframework.beans.factory.config.AutowireCapableBeanFactory;
+import org.springframework.beans.factory.support.BeanDefinitionRegistry;
+import org.springframework.beans.factory.support.GenericBeanDefinition;
+import org.springframework.boot.web.embedded.tomcat.TomcatServletWebServerFactory;
+import org.springframework.boot.web.servlet.ServletRegistrationBean;
+import org.springframework.boot.web.servlet.server.ServletWebServerFactory;
+import org.springframework.context.ApplicationContext;
+import org.springframework.context.ApplicationContextAware;
+import org.springframework.context.annotation.Bean;
+
+public abstract class CommonApp implements ApplicationContextAware {
```

Comment:
No we can not use `"favor composition over inheritance"` pattern here because the use cases is different. Here the matter is not about variables but it is about common methods implementation which is common for all other backend as well. and some methods are different but signature are same for all backend so the abstraction is the perfect design for this as per my point of view.

Reply:
If your goal is just to expose some spring beans, that doesn't need to be done via a class hierarchy at all. It can just be a class that exposes them, included in the packages that the application scans for startup.

Reply:
@jkwatson-verta Please consider follow up PR as well while reviewing because this PR is just the initial setup which might be expanded in follow up PRs like: https://github.com/VertaAI/modeldb/pull/3005/files#diff-49c276c2dafbfa6661d2997338e0594e9bbf58374e75622216d9a30b102dee87 so there will be abstract methods where it implementation is different for all other backends.

Reply:
I do not think this is a good design. Please reconsider using a base class for this.